File: src/main/java/twao/villages/TargetVillage.java
package twao.villages;

public class TargetVillage extends Village {
    private int numberOfAttacks;

    /**
     * @param x                 X coordinate
     * @param y                 Y coordinate
     * @param numberOfAttacks   The number of attacks for a given village
     */
    public TargetVillage(int x, int y, int numberOfAttacks) {
        super(x, y);
        this.numberOfAttacks = numberOfAttacks;
    }

    /**
     * Decreases {@code numberOfAttacks} by 1.
     */
    public void attack()                { numberOfAttacks--; }
    public int getNumberOfAttacks()     { return numberOfAttacks; }
    /**
     * @return {@code true} only if there is no more attacks needed
     */
    public boolean isAssignCompleted()  { return numberOfAttacks > 0 ? false : true; }
}

File: src/main/java/twao/villages/TargetVillage.kt
package twao.villages

data class TargetVillage(
        override val x: Int,
        override val y: Int,
        private var _numberOfAttacks: Int
) : Village(x, y) {
    val numberOfAttacks
        get() = _numberOfAttacks

    fun attack()
        { _numberOfAttacks-- }

    fun isAssignCompleted(): Boolean
        = _numberOfAttacks <= 0
}